src/dump.c

```diff
@@ -2810,7 +2810,7 @@ static void jl_recache_other(void)
             _new = (jl_value_t*)jl_recache_method_instance((jl_method_instance_t*)o, i);
         }
         else {
-            assert(0);
+            abort();
```

Comment:
If this didn't trigger a crash here it will cause crashes that's even harder to debug. (I'm not sure how that can happen actually, most likely memory corruption I guess)